server/src/server.ts

```diff
@@ -6,269 +6,84 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import * as ts from 'typescript/lib/tsserverlibrary';
-import * as lsp from 'vscode-languageserver';
-
-import {tsCompletionEntryToLspCompletionItem} from './completion';
 import {createLogger} from './logger';
-import {ProjectService} from './project_service';
 import {ServerHost} from './server_host';
-import {filePathToUri, lspPositionToTsPosition, lspRangeToTsPositions, tsTextSpanToLspRange, uriToFilePath} from './utils';
-
-enum LanguageId {
-  TS = 'typescript',
-  HTML = 'html',
+import {Session} from './session';
+import {resolveWithMinMajor} from './version_provider';
+
+// Parse command line arguments
+const help = hasArgument('--help');
+const logFile = findArgument('--logFile');
+const logVerbosity = findArgument('--logVerbosity');
+const ngProbeLocations = parseStringArray('--ngProbeLocations');
+const tsProbeLocations = parseStringArray('--tsProbeLocations');
+
+if (help) {
+  const {argv} = process;
+  console.error(`Angular service that implements Language Server Protocol (LSP).
+
+Usage: ${argv[0]} ${argv[1]} [options]
+
+Options:
+  --help: Prints help message.
+  --logFile: Location to log messages. Logging is disabled if not provided.
+  --logVerbosity: terse|normal|verbose|requestTime. See ts.server.LogLevel.
+  --ngProbeLocations: Location of @angular/language-service. Required.
+  --tsProbeLocations: Location of typescript. Required.
+
+Additional options supported by vscode-languageserver:
+  --node-ipc: Communicate using Node's IPC. This is the default.
+  --stdio: Communicate over stdin/stdout.
+  --socket=<number>: Communicate using Unix socket.
+`);
+  process.exit(0);
 }
 
-// Parse startup options
-const options = new Map<string, string>();
-for (let i = 0; i < process.argv.length; ++i) {
-  const arg = process.argv[i];
-  if (arg === '--logFile') {
-    options.set('logFile', process.argv[i + 1]);
-  } else if (arg === '--logVerbosity') {
-    options.set('logVerbosity', process.argv[i + 1]);
-  } else if (arg === '--pluginProbeLocation') {
-    options.set('pluginProbeLocation', process.argv[i + 1]);
-  }
-}
+// Create a logger that logs to file. OK to emit verbose entries.
+const logger = createLogger({logFile, logVerbosity});
 
-// Create a connection for the server. The connection uses Node's IPC as a transport
-const connection: lsp.IConnection = lsp.createConnection();
+const ts = resolveWithMinMajor('typescript', 3, tsProbeLocations)!;
```

Comment:
Do you mean an import statement `import * as ts from 'typescript` could be added later and it could result in a name conflict? TypeScript disallows that, it produces error `Import declaration conflicts with local declaration of 'ts'.ts(2440)`

Reply:
okay, cool.

I think also that the `!` check at the end of this line is not needed, the call to `resolveMinMajor` always returns a `NodeModule`.